File: CS/WinForms_MapControl_VecLayer_DataLoadedEvent/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using DevExpress.XtraMap;

namespace WinForms_MapControl_VecLayer_DataLoadedEvent {
    public partial class Form1 : Form {
        const string filepath = "..\\..\\Earthquakes.xml";

        VectorItemsLayer vectorLayer = new VectorItemsLayer();

        public Form1() {
            InitializeComponent();
            InitializeMap();
        }

        private void InitializeMap() {
            MapControl map = new MapControl() { Dock = DockStyle.Fill };
            this.Controls.Add(map);

            map.Layers.Add(new ImageTilesLayer() {
                DataProvider = new BingMapDataProvider() {
                    BingKey = "YOUR BING KEY"
                }
            });
            map.Layers.Add(vectorLayer);

            vectorLayer.DataLoaded += vectorLayer_DataLoaded;
        }

        #region #DataLoaded
        void vectorLayer_DataLoaded(object sender, DataLoadedEventArgs e) {
            MapItemsLoadedEventArgs args = e as MapItemsLoadedEventArgs;
            if (args == null) return;

            lLayerState.Text = args.Items.Count().ToString() + " items loaded.";
        }
        #endregion #DataLoaded

        IMapDataAdapter CreateData(object datasource) {
            BubbleChartDataAdapter adapter = new BubbleChartDataAdapter() { DataSource = datasource, ItemMaxSize = 50, ItemMinSize = 10 };
            adapter.Mappings.Latitude = "glat";
            adapter.Mappings.Longitude = "glon";
            adapter.Mappings.Value = "mag";

            return adapter;
        }

        private DataTable LoadData(string path) {
            DataSet ds = new DataSet();
            ds.ReadXml(path);
            DataTable table = ds.Tables[0];
            return table;
        }

        private void bLoadData_Click(object sender, EventArgs e) {
            lLayerState.Text = "Please wait.";
            vectorLayer.Data = CreateData(LoadData(filepath));
        }
    }
}

File: VB/WinForms_MapControl_VecLayer_DataLoadedEvent/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports DevExpress.XtraMap

Namespace WinForms_MapControl_VecLayer_DataLoadedEvent
    Partial Public Class Form1
        Inherits Form

        Private Const filepath As String = "..\..\Earthquakes.xml"

        Private vectorLayer As New VectorItemsLayer()

        Public Sub New()
            InitializeComponent()
            InitializeMap()
        End Sub

        Private Sub InitializeMap()
            Dim map As New MapControl() With {.Dock = DockStyle.Fill}
            Me.Controls.Add(map)

            map.Layers.Add(New ImageTilesLayer() With { _
                .DataProvider = New BingMapDataProvider() With {.BingKey = "YOUR BING KEY"} _
            })
            map.Layers.Add(vectorLayer)

            AddHandler vectorLayer.DataLoaded, AddressOf vectorLayer_DataLoaded
        End Sub

        #Region "#DataLoaded"
        Private Sub vectorLayer_DataLoaded(ByVal sender As Object, ByVal e As DataLoadedEventArgs)
            Dim args As MapItemsLoadedEventArgs = TryCast(e, MapItemsLoadedEventArgs)
            If args Is Nothing Then
                Return
            End If

            lLayerState.Text = args.Items.Count().ToString() & " items loaded."
        End Sub
        #End Region ' #DataLoaded

        Private Function CreateData(ByVal datasource As Object) As IMapDataAdapter
            Dim adapter As New BubbleChartDataAdapter() With {.DataSource = datasource, .ItemMaxSize = 50, .ItemMinSize = 10}
            adapter.Mappings.Latitude = "glat"
            adapter.Mappings.Longitude = "glon"
            adapter.Mappings.Value = "mag"

            Return adapter
        End Function

        Private Function LoadData(ByVal path As String) As DataTable
            Dim ds As New DataSet()
            ds.ReadXml(path)
            Dim table As DataTable = ds.Tables(0)
            Return table
        End Function

        Private Sub bLoadData_Click(ByVal sender As Object, ByVal e As EventArgs) Handles bLoadData.Click
            lLayerState.Text = "Please wait."
            vectorLayer.Data = CreateData(LoadData(filepath))
        End Sub
    End Class
End Namespace
